scripts/SANS/sans/gui_logic/models/diagnostics_page_model.py

```diff
@@ -0,0 +1,166 @@
+from sans.common.general_functions import parse_diagnostic_settings
+from mantid.simpleapi import SumSpectra, ConvertAxesToRealSpace
+from sans.common.general_functions import (create_child_algorithm)
+from sans.common.constants import EMPTY_NAME
+from sans.common.enums import IntegralEnum, DetectorType, SANSDataType
+from mantid.api import AlgorithmPropertyWithValue
+from sans.algorithm_detail.batch_execution import provide_loaded_data, create_unmanaged_algorithm, add_to_group
+from sans.common.file_information import get_instrument_paths_for_sans_file
+from sans.common.xml_parsing import get_named_elements_from_ipf_file
+from sans.gui_logic.models.table_model import TableModel, TableIndexModel
+from sans.gui_logic.presenter.gui_state_director import (GuiStateDirector)
+from mantid import AnalysisDataService
+
+try:
+    import mantidplot
+except (Exception, Warning):
+    mantidplot = None
+
+
+def run_integral(range, mask, integral, detector, state):
+    ranges = parse_range(range)
+    input_workspaces = load_workspace(state)
+
+    is_multi_range = len (ranges) > 1
+
+    output_workspaces = []
+    for input_workspace in input_workspaces:
+        input_workspace_name = input_workspace.name()
+        input_workspace = crop_workspace(DetectorType.to_string(detector), input_workspace)
+
+        if mask:
+            input_workspace = apply_mask(state, input_workspace, DetectorType.to_string(detector))
+
+        x_dim, y_dim = get_detector_size_from_sans_file(state, detector)
+
+        output_workspace = integrate_ranges(ranges, integral, mask, detector, input_workspace_name, input_workspace, x_dim, y_dim,
+                                            is_multi_range)
+        plot_graph(output_workspace)
+
+        output_workspaces.append(output_workspace)
+
+    return output_workspaces
+
+
+def integrate_ranges(ranges, integral, mask, detector, input_workspace_name, input_workspace, x_dim, y_dim, is_multi_range):
+    for range in ranges:
```

Comment:
`range` is a built-in function and being shadowed here. Actually, it seems to be shadowed in a lot of places below.